platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/jvm/DaemonJavaInstallationRegistryFactory.java

```diff
@@ -0,0 +1,92 @@
+import org.gradle.internal.jvm.inspection.JavaInstallationRegistry;
+import org.gradle.internal.jvm.inspection.JvmMetadataDetector;
+import org.gradle.internal.logging.progress.ProgressLoggerFactory;
+import org.gradle.internal.os.OperatingSystem;
+import org.gradle.jvm.toolchain.internal.AsdfInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.AutoInstalledInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.CurrentInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.InstallationSupplier;
+import org.gradle.jvm.toolchain.internal.IntellijInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.JabbaInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.LinuxInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.MavenToolchainsInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.OsXInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.SdkmanInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.WindowsInstallationSupplier;
+import org.gradle.jvm.toolchain.internal.install.JdkCacheDirectory;
+import org.gradle.process.internal.ExecFactory;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+import static org.gradle.jvm.toolchain.internal.AutoDetectingInstallationSupplier.AUTO_DETECT;
+import static org.gradle.jvm.toolchain.internal.AutoInstalledInstallationSupplier.AUTO_DOWNLOAD;
+
+public class DaemonJavaInstallationRegistryFactory implements JavaInstallationRegistryFactory {
+
+    private final JdkCacheDirectory jdkCacheDirectory;
+    private final JvmMetadataDetector jvmMetadataDetector;
+    private final ExecFactory execFactory;
+    private final ProgressLoggerFactory progressLoggerFactory;
+    private final WindowsRegistry windowsRegistry;
+
+    public DaemonJavaInstallationRegistryFactory(JdkCacheDirectory jdkCacheDirectory, JvmMetadataDetector jvmMetadataDetector, ExecFactory execFactory, ProgressLoggerFactory progressLoggerFactory, WindowsRegistry windowsRegistry) {
+        this.jdkCacheDirectory = jdkCacheDirectory;
+        this.jvmMetadataDetector = jvmMetadataDetector;
+        this.execFactory = execFactory;
+        this.progressLoggerFactory = progressLoggerFactory;
+        this.windowsRegistry = windowsRegistry;
+    }
+
+    @Override
+    @Nonnull
+    public JavaInstallationRegistry getRegistry(StartParameter startParameter) {
+        List<InstallationSupplier> installationSuppliers = getInstallationSuppliers(startParameter);
+        return new JavaInstallationRegistry(installationSuppliers, jvmMetadataDetector, null, OperatingSystem.current(), progressLoggerFactory);
+    }
+
+    private List<InstallationSupplier> getInstallationSuppliers(StartParameter startParameter) {
+        DaemonProviderFactory providerFactory = new DaemonProviderFactory(startParameter);
+        // The gradle property 'org.gradle.java.installations.auto-detect' must be ignored for Daemon Toolchain
```

Comment:
We [agreed](https://docs.google.com/document/d/1p0QohF1TAZOVWh9yCgsp97H6uDRAjHZeLNmD4dT1IGc/edit?disco=AAAA3nocE1M) for now to move this under future considerations after removed a section called  `Disable toolchain auto-detection`:
```
The existing auto-detection mechanism can be disabled using the “[org.gradle.java.installations.auto-detect](https://docs.gradle.org/current/userguide/toolchains.html#sub:disable_auto_detect)” option. However, this will be expanded to be applicable also for the Gradle toolchain
```

Similar like happens with `org.gradle.java.installations.fromEnv` and `org.gradle.java.installations.paths` those properties needs to be defined under `gradle.properties` which creates confusion with the existing implementation and the one for the daemon toolchain if we reuse exactly the same.